README.md

```diff
@@ -1,38 +1,239 @@
+
+```json
+"search-result": {
+    "allowed": [
+      "not-found",
+      "breadcrumb",
+      "filter-navigator",
+      "total-products",
+      "order-by"
+    ],
+    "required": [
+      "gallery"
+    ],
+    "component": "index"
+  },
+```
+
+The search-result has as a required block the `gallery`. So, any search-result block implementation created must add a gallery as a block that is inside of search-result. Similarly, `gallery` has its own inner block structure that can be configured that you can see below.
+
+```json
+ "gallery": {
+    "required": [
+      "product-summary"
+    ],
+    "component": "Gallery"
+  }
+```
+
+The gallery has as a required block the `product-summary`. So, any gallery block implementation created must add a product-summary as a block that is inside of gallery. (Similarly, `product-summary` has its own inner block structure that can be configured. There is a link to its API in the next section.)
+
+### Configuration
+
+Through the Storefront, you can change the search-result behavior and interface. However, you also can make in your theme app, as Dreamstore does.
 
-To run the test suit, type the following in the terminal, inside the folder `react`
+| Prop name          | Type       | Description   | Default value |
+| ------------------ | ---------- | --------------------------------------------------------------------------- | --- |
+| `querySchema`                      | `QuerySchema`   | Query made when there's no context                     | N/A
+| `hiddenFacets`          | `HiddenFacets`  | Indicates which facets will be hidden                             | N/A
+| `pagination`                   | `String`  | Pagination type (values: 'show-more' or 'infinite-scroll')       | `infinity-scroll`
 
-```sh
-$ npm t     # this
-$ npm test  # or
+QuerySchema
+
+| Prop name          | Type       | Description  | Default value |
+| ------------------ | ---------- | --------------------------------------------------------------------------- | --- |
+| `maxItemsPerPage`   | `Number`   | Maximum number of items per search page                     | 10 |
+| `queryField`          | `String`  | Query field                             | N/A
+| `mapField`              | `String`  | Map field                                                  | N/A |
+| `restField`                   | `String`  | Other Query Strings       | N/A |
+| `orderByField`                   | `String`  | Order by field (values: 'OrderByTopSaleDESC', 'OrderByReleaseDateDESC', 'OrderByBestDiscountDESC', 'OrderByPriceDESC', 'OrderByPriceASC', 'OrderByNameASC' or 'OrderByNameDESC')       | `OrderByReleaseDateDESC` |
+ 
+HiddenFacets
+
+| Prop name          | Type       | Description   | Default value |
+| ------------------ | ---------- | --------------------------------------------------------------------------- | --- |
+| `layoutMode1`            | `String`  | Layout mode of the switcher (values: 'normal', 'small' or 'inline')    | `normal` |
```

Comment:
Isn't this a `Enum` type?

Reply:
No, it is a String. :smile:

Reply:
Isn't is a `Enum` of type String?

In this case 
```suggestion
| `layoutMode1`            | `Enum`  | Layout mode of the switcher (Possible values: 'normal', 'small' or 'inline')    | `normal` |
```